project/project_test.go

```diff
@@ -349,6 +349,98 @@ func TestImportTasks(t *testing.T) {
 
 }
 
+func TestPresets(t *testing.T) {
+	unmarshal := func(s string) project.Parameters {
+		var p project.Parameters
+		if err := yaml.Unmarshal([]byte(s), &p); err != nil {
+			t.Fatal(err)
+		}
+		return p
+	}
+	fs.SetContent("bar://foo.ttcn3", []byte(`module foo {
+    type component C0 {}
+    testcase TC1() runs on C0 system C0 {}
+}`))
+	a := unmarshal(`
+                timeout: 0
+                presets:
+                   "A":
+                      timeout: 1
+                      parameters:
+                          "mod1.p1": 1
+                   "B":
+                      timeout: 2
+                execute:
+                  - test: "foo.TC1"
+                    timeout: 3
+                    except:
+                        presets:
+                        - A
+                  - test: "foo.TC1"
+                    timeout: 4
+                    only:
+                         presets:
+                         - A`)
+
+	t.Run("Preset_A", func(t *testing.T) {
+		expected := unmarshal(`
+        execute:
+          - test: "foo.TC1"
+            timeout: 4
+            only:
+                 presets:
+                 - A`)
+		var p []project.TestConfig
+		p = append(p, project.AcquireExecutables(&a, []string{"bar://foo.ttcn3"}, []string{"A"})...)
+		assert.Equal(t, &expected, &project.Parameters{Execute: p})
+	})
+
+	t.Run("Preset_B", func(t *testing.T) {
+		expected := unmarshal(`
+        execute:
+          - test: "foo.TC1"
+            timeout: 3
+            except:
+                 presets:
+                 - A`)
+		var p []project.TestConfig
+		p = append(p, project.AcquireExecutables(&a, []string{"bar://foo.ttcn3"}, []string{"B"})...)
+		assert.Equal(t, &expected, &project.Parameters{Execute: p})
+	})
+
+	t.Run("No_Preset", func(t *testing.T) {
+		expected := unmarshal(`
+        execute:
+          - test: "foo.TC1"
+            timeout: 3
+            except:
+                 presets:
+                 - A`)
+		var p []project.TestConfig
+		p = append(p, project.AcquireExecutables(&a, []string{"bar://foo.ttcn3"}, []string{})...)
+		assert.Equal(t, &expected, &project.Parameters{Execute: p})
+	})
+
+	t.Run("Preset_AB", func(t *testing.T) {
+		expected := unmarshal(`
+        execute:
+        - test: "foo.TC1"
+          timeout: 3
+          except:
+            presets:
+            - A
+        - test: "foo.TC1"
+          timeout: 4
+          only:
+            presets:
+            - A`)
+		var p []project.TestConfig
+		p = append(p, project.AcquireExecutables(&a, []string{"bar://foo.ttcn3"}, []string{"B", "A"})...)
+		assert.Equal(t, &expected, &project.Parameters{Execute: p})
+	})
+
+}
+
```

Comment:
I would add some corner-case-tests. For example, what happens for those configurations:
```yaml
# empty only condition
- test: tc1
  only:

# empty except condition
- test: tc2
  except:

# non-existing only condition
- test: tc3
  only:
    presets: [ "does-not-exist" ]

# conflicting conditions
- test: tc4
  only:
    presets: [ "A" ]
  except:
    presets: ["A"]

# conflicting conditions alternate order
- test: tc5
  except:
    presets: ["A"]
  only:
    presets: ["A"]

```